lib/AST/TypeCheckRequests.cpp

```diff
@@ -429,3 +430,52 @@ void USRGenerationRequest::noteCycleStep(DiagnosticEngine &diags) const {
   auto &d = std::get<0>(storage);
   diags.diagnose(d, diag::circular_reference);
 }
+
+//----------------------------------------------------------------------------//
+// DefaultTypeRequest.
+//----------------------------------------------------------------------------//
+
+void swift::simple_display(llvm::raw_ostream &out,
+                           const DeclContextWrapper &x) {
+  if (!x.dc) {
+    out << "(null)";
+    return;
+  }
+  x.dc->printContext(out);
```

Comment:
`printContext()` prints out multiple lines, which makes it not work out well with (e.g.) the debugging dumps provided by the request-evaluator when diagnosing cycles. Can we instead use a more compact (and lossy) single-line format here?

Reply:
Do you know of one offhand, or should I write one?

Reply:
I've enhanced `printContext` with an `includeAncestors` flag, set to false when called by simple_display. I *think* that might do the trick.